### PR TITLE
Improve the Project structure and Resources section in flutter-for-android

### PR DESCRIPTION
**Pull request by rock3r**
This PR is the next in a series to improve the contents of `flutter-for-android.md`:

 * Fixing typos
 * Properly marking code in paragraphs as such (e.g., StatelessWidget -> `StatelessWidget`) to conform with the general conventions of the Flutter documentation, and for clarity
 * Improving sentence structure
 * Rewriting passages that are too vague, inexact, or could be misinterpreted by readers

Previous PRs: #927 #932 #933 (all required for this to be mergeable)

### CODE REVIEW

**Comment by rock3r** [comment]
# ❌ DO NOT MERGE ❌ 

Requires #927, #932 and #933 to be merged first!

File: flutter-for-android.md [hunk 1]
```diff
@@ -1000,7 +1021,7 @@ dependencies:
   http: '>=0.11.3+12'
```
Comment (mjohnsullivan) [inline flutter-for-android.md]: 0.11.3+16

**Comment by mjohnsullivan** [comment]
LGTM with tiny nit

File: flutter-for-android.md (hunk 1 above)
Comment (rock3r) [reply flutter-for-android.md]: 9a6adb53d2e7e72b569f5d2eb6ae3a7a935682ef

**Pull request merged**